agent/agent-bootstrap/src/main/java/com/microsoft/applicationinsights/agent/bootstrap/configuration/ConfigurationBuilder.java

```diff
@@ -292,4 +282,27 @@ private void log(Logger logger) {
             logger.warn(message, args);
         }
     }
+
+    public static Configuration loadJsonConfigFile(Path configPath) throws IOException{
+        if (!Files.exists(configPath)) {
+            throw new IllegalStateException("config file does not exist: " + configPath);
+        }
+        
+        BasicFileAttributes attributes = Files.readAttributes(configPath, BasicFileAttributes.class);
+        Long lastModifiedTime = attributes.lastModifiedTime().toMillis();
```

Comment:
```suggestion
        // important to read last modified before reading the file, to prevent possible race condition
        // where file is updated after reading it but before reading last modified, and then since
        // last modified doesn't change after that, the new updated file will not be read afterwards
        Long lastModifiedTime = attributes.lastModifiedTime().toMillis();
```